pkg/descheduler/descheduler.go

```diff
@@ -75,7 +76,21 @@ func Run(ctx context.Context, rs *options.DeschedulerServer) error {
 		<-ctx.Done()
 		close(stopChannel)
 	}()
-	return RunDeschedulerStrategies(ctx, rs, deschedulerPolicy, evictionPolicyGroupVersion, stopChannel)
+
+	runFn := func() error {
+		return RunDeschedulerStrategies(ctx, rs, deschedulerPolicy, evictionPolicyGroupVersion, stopChannel)
+	}
+
+	// if deployed as a Deployment,
+	if rs.LeaderElection.LeaderElect {
+		if rs.DeschedulingInterval.Seconds() > 0 {
+			leader.NewLeaderElection(runFn, rsclient, rs.LeaderElection, ctx)
+			return nil
+		}
+		klog.V(1).InfoS("Warning: DeschedulingInterval must be greater than 0 to use LeaderElection")
```

Comment:
Would you mind returning an error here? If the leader election is enabled but can not proceed, we need to error and tell the user to set the descheduling interval.